### DISCUSSION TITLE
How do I preview a single qmd file in VSCode without creating contents that are visible to git version control?

### DISCUSSION THREAD

**Discussion by pipme** [quarto-dev/quarto-cli#11388] [Q&A] [answered]
### Description

Hi, I am using Quarto for a `book` type project. In VSCode, if clicking the `preview` button on the upper right, there will be outputs like `<filename>_files/` and also `<filename>.html` (see the figure below). Can I direct these outputs to be inside a folder, e.g., `_preview_cache/`, such that I could add the folder to `.gitignore`?
<img width="361" alt="image" src="https://github.com/user-attachments/assets/84f7f7a9-78a5-440b-94ff-769dd3cbad17">

I tried:
```yaml
project:
  type: book
  preview:
    output-dir: "_preview_cache"
```
but it doesn't work.

Thanks! 

My Quarto version is 1.5.57.

**Comment by mcanouil** [comment]
Because `output-dir` belongs to `project` not `preview`.
I am not sure what you are trying to do exactly here.
Do you want to track the output at all?

**Reply by pipme** [reply]
Hi, I don't want to track the output. I want to either exclude the output in `.gitignore` or automatically delete them after previewing.

**Reply by mcanouil** [reply]
Then you don't need to change the output directory.
Simply add `_book` to your `.gitignore`.

Note that you are talking about single file but you are using a non default project which is a book.
So what is your setup exactly?
`output-dir` is a project option. There is a discussion about this.
- https://github.com/quarto-dev/quarto-cli/discussions/2171

Anyhow, use `.gitignore` to ignore what you don't want to track.

**Reply by pipme** [reply]
But the output is not inside `_book`. They are in the root folder.
<img width="348" alt="image" src="https://github.com/user-attachments/assets/c540fffd-a2ea-4c07-972d-25520cbe661b">
I guess that's because it gets treated as a single qmd file but not a file belonging to the book project.

**Reply by mcanouil** [reply]
Because you don't share concrete and reproducible example, I can only guess you are mixing incorrect things together hoping it will somehow work.

**Reply by pipme** [reply, marked as answer]
Thanks for the help! 

I just figured it out. The file `mlcs_guide.qmd` was not added to `.quarto.yaml`. After adding it there the output is corrected sent to `_book`. Sorry for bothering 😅

**Reply by pipme** [reply]
And yeah I got confused whether the extension is previewing a single file or a file inside a book project. Thanks!

**Reply by mcanouil** [reply]
Glad you figured it out!
